lib/foreman/renderer/scope/macros/host_template.rb

```diff
@@ -121,6 +121,62 @@ def ks_console
             (@port && @baud) ? "console=ttyS#{@port},#{@baud}" : ''
           end
 
+          apipie :method, 'Generates commands for package(s) installation' do
+            description 'Supported OS families: Redhat, Debian & Suse'
+            required :packages, String, desc: "Package(s) to install"
+            returns String, desc: 'Installation commands'
+            example "CentOS host: install_packages('pkg1') #=> yum -y install pkg1"
+            example "Fedora host: install_packages('pkg1 pkg2') #=> dnf -y install pkg1 pkg2"
+          end
+          def install_packages(packages)
+            return '' if packages.blank?
+
+            banner = <<~CMD
+              echo '#'
+              echo '# Installing packages'
+              echo '#'
+            CMD
+
+            case host.operatingsystem&.family
+            when 'Redhat'
+              os = host.operatingsystem
+              is_fedora = os.name.downcase == 'fedora'
+              is_rhel = os.name.downcase == 'redhat'
+
+              if is_fedora
+                return <<~CMD
+                  #{banner}
+                  #{os.major.to_i >= 22 ? 'dnf' : 'yum'} -y install #{packages}
+                CMD
+              end
+
+              if is_rhel
+                return <<~CMD
+                  #{banner}
+                  #{os.major.to_i >= 8 ? 'dnf' : 'yum'} -y install #{packages}
+                CMD
+              end
+
+              "yum -y install #{packages}"
```

Comment:
Based on my comment I would not have an explicit test for RHEL. Rather:
```suggestion
           is_dnf = (is_fedora && os.major.to_i >= 22) || (!is_fedora && os.major.to_i >= 8)
           <<~CMD
             #{banner}
             #{is_dnf ? 'dnf' : 'yum'} -y install #{packages}
           CMD
```
```